File: Back/NodeJS/modules/shared/infrastructure/uow/InMemoryUnitOfWork.ts
import { IUnitOfWork } from "./IUnitOfWork";
export class InMemoryUnitOfWork implements IUnitOfWork {
    Rollback(): Promise<void> {
        return Promise.resolve();
    }
    Commit(): Promise<void> {
        return Promise.resolve();
    }
    Begin(): Promise<void> {
        return Promise.resolve();
    }

}

File: Back/.Net/Modules/Users/BetFriends.Users.Infrastructure/UoW/InMemoryUnitOfWork.cs
using BetFriends.Shared.Infrastructure.Event;
using BetFriends.Shared.Infrastructure.UoW;

namespace BetFriends.Users.Infrastructure.UoW;

internal class InMemoryUnitOfWork : IUnitOfWork
{
    private readonly IDomainEventDispatcher _domainEventDispatcher;
    public InMemoryUnitOfWork(IDomainEventDispatcher domainEventDispatcher)
    {
        _domainEventDispatcher = domainEventDispatcher;
    }
    public Task Begin()
    {
        return Task.CompletedTask;
    }

    public async Task Commit()
    {
        await _domainEventDispatcher.DispatchAsync();
    }

    public Task Rollback()
    {
        return Task.CompletedTask;
    }
}
